server/lib/exporters/bag.py

```diff
@@ -0,0 +1,190 @@
+from datetime import datetime, timezone
+from hashlib import sha256, md5
+import json
+from girder.models.folder import Folder
+from girder.utility import JsonEncoder
+from . import TaleExporter
+
+
+bag_profile = (
+    "https://raw.githubusercontent.com/fair-research/bdbag/"
+    "master/profiles/bdbag-ro-profile.json"
+)
+bag_info_tpl = """Bag-Software-Agent: WholeTale version: 0.7
+BagIt-Profile-Identifier: {bag_profile}
+Bagging-Date: {date}
+Bagging-Time: {time}
+Payload-Oxum: {oxum}
+"""
+
+run_tpl = r"""#!/bin/sh
+
+# Use repo2docker to build the image from the workspace
+docker run  \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  -v `pwd`/data/workspace:/WholeTale/workspace \
+  --privileged=true \
+  -e DOCKER_HOST=unix:///var/run/docker.sock \
+  wholetale/repo2docker \
+  jupyter-repo2docker \
+    --target-repo-dir=/WholeTale/workspace \
+    --template={template} \
+    --buildpack-name={buildpack} \
+    --user-id=1000 --user-name={user} \
+    --no-clean --no-run --debug \
+    --image-name wholetale/tale_{taleId} \
+    /WholeTale/workspace
+
+docker run --rm \
+    -v `pwd`:/bag \
+    -ti jfloff/alpine-python:2.7-slim \
+      -p bdbag -- bdbag --resolve-fetch all /bag
+
+echo "========================================================================"
+echo " Open your browser and go to: http://localhost:{port}/{urlPath} "
+echo "========================================================================"
+
+# Run the built image
+docker run -p {port}:{port} \
+  -v `pwd`/data/data:/WholeTale/data \
+  -v `pwd`/data/workspace:/WholeTale/workspace \
+  wholetale/tale_{taleId} {command}
+
+"""
+
+
+readme_tpl = """# Tale: "{title}" in BDBag Format
+
+{description}
+
+# How to run?
+
+```
+sh ./run.sh
+```
+
+Access on http://localhost:{port}/{urlPath}
+"""
+
+
+class BagTaleExporter(TaleExporter):
+    def stream(self):
+        token = 'wholetale'
+        container_config = self.image['config']
+        rendered_command = container_config.get('command', '').format(
+            base_path='', port=container_config['port'], ip='0.0.0.0', token=token
+        )
+        urlPath = container_config['urlPath'].format(token=token)
+        run_file = run_tpl.format(
+            template=container_config['template'],
+            buildpack=container_config['buildpack'],
+            user=container_config['user'],
+            port=container_config['port'],
+            taleId=self.tale['_id'],
+            command=rendered_command,
+            urlPath=urlPath,
+        )
+        top_readme = readme_tpl.format(
+            title=self.tale['title'],
+            description=self.tale['description'],
+            port=container_config['port'],
+            urlPath=urlPath,
+        )
+        extra_files = {
+            'data/LICENSE': self.tale_license['text'],
+            'README.md': top_readme,
+            'run.sh': run_file,
```

Comment:
I think we should rename this `run-local.sh` or similar to avoid confusion with the `run.sh` we talk about in the workspace for running the actual analysis.

Reply:
https://github.com/whole-tale/girder_wholetale/pull/271/commits/2efc8b6feaada2859b770f672ab1a254f76337a8